rustler_codegen/src/nif.rs

```diff
@@ -0,0 +1,205 @@
+use proc_macro2::{Span, TokenStream};
+use quote::{quote, ToTokens};
+use syn::punctuated::Punctuated;
+use syn::token::Comma;
+
+pub fn transcoder_decorator(args: syn::AttributeArgs, fun: syn::ItemFn) -> TokenStream {
+    let sig = &fun.sig;
+    let name = &sig.ident;
+    let inputs = &sig.inputs;
+    let flags = schedule_flag(args.to_owned());
+    let function = fun.to_owned().into_token_stream();
+    let arity = arity(inputs.clone());
+    let decoded_terms = extract_inputs(inputs.clone());
+    let argument_names = create_function_params(inputs.clone());
+    let erl_func_name = extract_attr_value(args.clone(), "name")
+        .map(|ref n| syn::Ident::new(n, Span::call_site()))
+        .unwrap_or_else(|| name.clone());
+
+    quote! {
+        #[allow(non_camel_case_types)]
+        pub struct #name;
```

Comment:
This could also be `pub enum #name{}`. With that, it is impossible to create an instance of that type. As the type is "only" used to implement `rustler::Nif` on it, and the type is not passed around, this could make sense.

Reply:
This did not work.